tests/integration/test_legacy_mysql.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.helpers import (
+    execute_queries_on_unit,
+    get_server_config_credentials,
+    get_unit_address,
+    scale_application,
+)
+
+logger = logging.getLogger(__name__)
+
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+CLUSTER_NAME = "test_cluster"
+DATABASE_APP_NAME = "mysql"
+OSM_KEYSTONE_APP_NAME = "osm-keystone"
+KFP_API_APP_NAME = "kfp-api"
+
+
+@pytest.mark.order(1)
+@pytest.mark.skip_if_deployed
+@pytest.mark.legacy_mysql_tests
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy the mysql charm.
+
+    Args:
+        ops_test: The ops test framework
+    """
+    async with ops_test.fast_forward():
+        # Build and deploy the mysql charm
+        charm = await ops_test.build_charm(".")
+        resources = {"mysql-image": METADATA["resources"]["mysql-image"]["upstream-source"]}
+        config = {
+            "mysql-interface-user": "keystone",
+            "mysql-interface-database": "keystone",
+        }
+        await ops_test.model.deploy(
+            charm,
+            resources=resources,
+            config=config,
+            application_name=DATABASE_APP_NAME,
+            num_units=3,
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[DATABASE_APP_NAME],
+            status="active",
+            raise_on_blocked=True,
+            timeout=1000,
+            wait_for_exact_units=3,
+        )
+        assert len(ops_test.model.applications[DATABASE_APP_NAME].units) == 3
+        for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+            assert unit.workload_status == "active"
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.legacy_mysql_tests
+async def test_osm_keystone_bundle_mysql(ops_test: OpsTest) -> None:
+    """Deploy the osm keystone bundle to test the legacy 'mysql' relation.
```

Comment:
this function is confusing for a noob like me, can you be more specific about what is exactly getting tested by the osm keystone bundle here?